be1-go/popcha/server.go

```diff
@@ -328,6 +344,56 @@ func (as *AuthorizationServer) generateQRCode(w http.ResponseWriter, req *http.R
 	return nil
 }
 
+// helper method used for handling redirect uri by the Webpage websocket. Requests on the /response endpoint are
+// solely. handled through websocket. Because of the protocol, the first websocket to send a request will be the JS
+// one.
+func (as *AuthorizationServer) responseEndpoint(w http.ResponseWriter, r *http.Request) {
```

Comment:
For someone who's not familiar with the protocol, it's a bit hard to understand what the method does exactly as it has many nested conditions etc ... Writing one or two helper methods for it (for example one for the first if/else and another one for the rest) with descriptive names and documentations can make it more readable.